### DISCUSSION TITLE
Switch to v1.26 by default? How to switch existing cluster to new major release?

### DISCUSSION THREAD

**Discussion by aleksasiriski** [kube-hetzner/terraform-hcloud-kube-hetzner#700] [Q&A] [answered]
Since Kubernetes v1.27 is coming out next week, maybe we should consider switching to v1.26 by default?

Also a genuine question, is it possible to update k3s from 1.25 to 1.26 without recreating the whole cluster, or is it advised against?

**Comment by mysticaltech** [comment]
@aleksasiriski Just change the initial_k3s_channel, and if not reflected (I do not not remember), [edit the system-upgrade manifests](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/discussions/381#discussioncomment-4089894) in the cluster to change that value. As soon as you do, node by node, it will upgrade (or downgrade) like a charm! If you haven't seen it yet, it's worth the popcorn! I usually have a look with `watch -n 1 kubectl get nodes`,

**Reply by mysticaltech** [reply, marked as answer]
And it's really safe from what I have seen, unless you are running stuff like rancher that is not compatible with edge versions. About the version change, it's all about Rancher, but now it seems Rancher supports v1.26, so we are good to go, will do it! 👍

**Reply by codeagencybe** [reply]
@mysticaltech 
Does Rancher now also support the autoscaling feature? Because I understood this was a problem in versions < 1.25?

**Reply by codeagencybe** [reply]
@mysticaltech 
Are you sure about that v1.26 support in rancher? 

I checked the official docs and they still say only up to 1.24 :/ 
AS feature is only support from v1.25 so this means, with Rancher locked at 1.24, AS is still not possible? 

Did somebody already try in production and can confirm that the combination works fine? No weird problems? 

<img width="820" alt="image" src="https://user-images.githubusercontent.com/7568546/230353642-0eb762de-8880-47b4-af1b-2204f65e3dba.png">

https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-7-1/

**Reply by mysticaltech** [reply]
@codeagencybe I do not know. Best thing is to try! Remember, you to not have to rely on the default version. Please try v1.26 and let me know if it works please, if it doesn't work, you can always set the initial_k3s_channel to v1.25 or v1.24.

**Reply by aleksasiriski** [reply]
> @aleksasiriski Just change the initial_k3s_channel, and if not reflected (I do not not remember), [edit the system-upgrade manifests](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/discussions/381#discussioncomment-4089894) in the cluster to change that value. As soon as you do, node by node, it will upgrade (or downgrade) like a charm! If you haven't seen it yet, it's worth the popcorn! I usually have a look with `watch -n 1 kubectl get nodes`,

`terraform apply` doesn't work because plans for server and agent are missing `metadata.resourceVersion: "your_clusters_ver"`, I had to add that by getting it from my cluster and it worked like a charm. Could that be automated somehow?

**Reply by mysticaltech** [reply]
@aleksasiriski I think just adding initial_k3s_channel to the `null_ressource.kustomization` would fix this automatically. PR welcome! 🙏

**Reply by aleksasiriski** [reply]
> @aleksasiriski I think just adding initial_k3s_channel to the `null_ressource.kustomization` would fix this automatically. PR welcome! 🙏

I'm testing it now

It works now!
